samples/java-protobuf-tracing/README.md

```diff
@@ -0,0 +1,60 @@
+## This example show how to create Spans by the users
+ 
+
+
+## Running Locally
+
+
+When running a Kalix service locally, we need to have its companion Kalix Runtime running alongside it.
+
+To start your service locally, run:
+
+```shell
+mvn kalix:runAll
+```
+
+It's worth noting that `application.conf` is passing `kalix.telemetry.tracing.collector-endpoint="http://localhost:4317"` 
```

Comment:
What happens if we deploy this application in Kalix? Will it read that property and start to call localhost:4317?

Maybe better to pass it to the mvn process as a `-D` property instead of having it in the `application.conf`. We can also pass `-D` properties in the pom.xm, I believe.

Reply:
Added as a `jvmArgs`. By the way there's not docs about that. How about we create some?

Reply:
I don't know. It depends I guess. 

Passing jvmArgs to main class is part of the basic skills set that a developer building a jvm application needs to have. If we need to document each single bit, then the docs becomes bloated with a lot of non-kalix information.

We can add a comment on the pom explaining why we passing the args. Same as in the README file.

What might be useful is a short notice on how to add tracing spans in dev-mode pointing to the sample. I mean, a short notice in the docs section that talks about adding tracing spans. Similar to what we have about enabling ACL in dev-mode.

Reply:
I mean that the `jvmArgs` as a property is not a pom.xml standard AFAIK. You can add something similar (with a different name) to the compiler-maven-plugin or the failsafe one for example. As we are using the exec-maven-plugin under the covers I understand that plugin has not `jvmArgs` as such. With that exact name.

Reply:
I exec-maven-plugin had such a param we could just state we are using that under the cover. On that I totally agree.

Reply:
It's not clear to me what you want to document exactly. Are you referring to the fact that someone can pass any `-D` args to the exec-maven-plugin? 

My point was more about the sample where we explain why we are passing this param in the pom.

And then, in the docs page about tracing, we should mention that one needs to add an extra container (jaeger) and let Kalix know about it.

By the way, I just realised that the same property needs to be added to the runtime container. So maybe we should not add anything to the pom.xml at all. 

The kalix plugin is already able to read the docker-compose file and pick the `-D` properties that we pass to the runtime and pass it to the jvm process as well. 
We don't do it for all the properties though, only for the service mappings. So instead, we could do the same for this new property and let the plugin solve it for the user automatically. 

This has the advantage that if you start kalix from inside the IDE (for example when debugging), it will also read the docker-compose file and pick the `-D` property.

Reply:
This is the dev-mode class that reads the docker-compose and add extra properties to the jvm process.
https://github.com/lightbend/kalix-jvm-sdk/blob/027d68229b5ad9b0c2284397b2b7bb5102ff052c/devtools/src/main/scala/kalix/devtools/impl/DevModeSettings.scala#L64

This is also used when starting from inside the IDE

Reply:
Oh, I'm wrong about it. It's not the same config. 

Hmm, the problem now is that when debugging the application inside the IDE, we won't have access to this property and telemetry will be out of the game. Unless the user explicitly pass it to their debug configuration inside the IDE.

Reply:
darn. Well, we could provide this for now as suggested https://github.com/lightbend/kalix-jvm-sdk/pull/2051#discussion_r1538988745 and then  in another PR try to add this to the debug.

Reply:
That won't work as suggested because the property is not the same. Nor the host. 
I think we could go now with the property in pom.xml and then in another PR try to improve the DX by reading the docker-compose file and transforming the telemetry property we pass to the runtime to something the user service can consume. 

But that should be in another PR. 

And we don't need to document anything for now, because if we achieve good DX in follow-up PR, nothing needs to be documented. 😄